### PR TITLE
Closes #168: IQ Skill restrictions

### LINKED ISSUE #168: IQ skill restrictions
Split from #18
> Oh and the IQ skill restrictions are also inside arm9.bin (0xa1fe0 -> 0xa2069)
> Each skill has a 2-byte "restriction group" value. If two skills have the same restriction group, they are incompatible and can't be enabled at the same time

US offset: 0xA1A5C

### PR DESCRIPTION
**Pull request by theCapypara**
Closes #168

I have the feeling the data isn't correct yet? I'm not sure where I messed up, can you check?

See also skytemple/skytemple#331

### CODE REVIEW

File: skytemple_files/hardcoded/iq.py [hunk 1]
```diff
@@ -153,18 +150,40 @@ def set_nectar_gain(value: int, ov29: bytearray, config: Pmd2Data):
...
+            # Null entry not present in restriction table
+            if i == 0:
+                lst.append(IqSkill(
+                    read_sintle(arm9bin, block.begin + (i * IQ_SKILL_ENTRY_LEN), 4),
+                    None
+                ))
+            else:
+                lst.append(IqSkill(
+                    read_sintle(arm9bin, block.begin + (i * IQ_SKILL_ENTRY_LEN), 4),
+                    read_sintle(arm9bin, block.begin + ((i - 1) * IQ_SKILL_RESTR_ENTRY_LEN), 2)
```
Comment (End45) [inline skytemple_files/hardcoded/iq.py:166]:
```suggestion
                    read_sintle(arm9bin, block_restr.begin + ((i - 1) * IQ_SKILL_RESTR_ENTRY_LEN), 2)
```

Comment (theCapypara) [reply skytemple_files/hardcoded/iq.py:166]: there we go :)

**Review by End45 — CHANGES_REQUESTED** [review]
I don't know if it's an UI bug, but the restriction values are off by 1. The inital 9999 corresponds to IQ skill #0.

**Comment by theCapypara** [comment]
You are right. The list had one entry less back when I used the wrong table for data which is why I thought it just didn't have a null entry. Fixed.

**Review by End45 — APPROVED** [review]
Looks good now

**Pull request merged**